test/test_stream_classes_model.rb

```diff
@@ -87,3 +87,71 @@ def self.startup
     }
   end
 end
+
+class TestSourceDetailsComparisonAllTypes < Test::Unit::TestCase
```

Comment:
I think we can refractor this code to use the `module` trick for having a general tests

Reply:
Now, I can not imagine how, but lets discuss that, I think you can enlighten me !!